biogtr/data_structures.py

```diff
@@ -16,6 +16,7 @@ def __init__(
         pred_track_id: int = -1,
         bbox: ArrayLike = torch.empty((0, 4)),
         crop: ArrayLike = torch.tensor([]),
+        centroid: dict[str, ArrayLike] = None,
```

Comment:
Bad! As is initializing with the `torch.empty` and `torch.tensor` above since these are all mutable!

Just change the type to `centroid: Optional[dict[str, ArrayLike]] = None` or `centroid: dict[str, ArrayLike] | None = None` (newer python syntax)

Reply:
I opened an issue to go through and remove all the mutable defaults.